src/mgmt/cli_mgmt.c

```diff
@@ -371,16 +371,28 @@ get_attach_info(const char *name, int *npsrs, struct dc_mgmt_psr **psrs,
 
 	if (sy_info) {
 		size = sizeof(sy_info->provider);
+		if (strlen(resp->provider) >= size) {
+			D_ERROR("GetAttachInfo provider string too long\n");
+			rc = -DER_INVAL;
+			goto out_resp;
```

Comment:
Could use the macro `D_GOTO(out_resp, rc = -DER_INVAL)` to condense these two lines. I wouldn't block on that though.

Reply:
I made the change.